pkg/archive/archive.go

```diff
@@ -382,6 +382,56 @@ func createTarFile(path, extractDir string, hdr *tar.Header, reader io.Reader, L
 	return nil
 }
 
+// Matches returns true if relFilePath matches any of the patterns
+// and isn't excluded by any of the subsequent patterns.
+func Matches(relFilePath string, patterns []string) (bool, error) {
+
+	if filepath.Clean(relFilePath) == "." {
```

Comment:
Because this is here, and not in the inner-loop after the removal of the "!", we may need to check for "!." here.